### ISSUE TITLE
setItemSpec -> setItem using wrong param?

### ISSUE THREAD

**Issue by aljachimiak** [oddnetworks/oddworks#213]
A recent merge made some errors more apparent in testing.  While tests pass this message appears.... a lot:
````
Unhandled rejection Error: spec.source String is required
    at setItem (/Users/aljachimiak/Projects/odd/al-odd/lib/services/catalog/commands/index.js:20:19)
    at commandHandlerWrapper (/Users/aljachimiak/Projects/odd/al-odd/node_modules/oddcast/lib/bus.js:84:16)
    at Promise._execute (/Users/aljachimiak/Projects/odd/al-odd/node_modules/bluebird/js/release/debuggability.js:272:9)
...
````
## Steps to Reproduce (for bugs)

(1)I added a source to all the items in the admin-speclist-spec

(2)I ran jasmine only on the admin-spec-list test.
`node_modules/jasmine/bin/jasmine.js spec/services/catalog/controllers/admin-spec-list-controller-spec.js`

(3) I _still_ got this error: `spec.source String is required`

(4) with some carefully placed console.log action, I noticed that the error, coming from `setItem`, was not being called directly by me, but by the `setItemSpec` method I was calling in the test.

## Current Behavior
When I call[ `bus.sendCommand({role: 'catalog', cmd: 'setItemSpec', type: 'collection'}, COLLECTION_SPEC),`](https://github.com/oddnetworks/oddworks/blob/master/spec/services/catalog/controllers/admin-spec-list-controller-spec.js#L104) in the test... we hit the correct method.

At the end of the `seItemSpec` method, [we call `setItem`](https://github.com/oddnetworks/oddworks/blob/master/lib/services/catalog/commands/index.js#L111-L114).

The problem is that the args setup in `setItemSpec` look like this:
````javascript
object = _.cloneDeep(sourceObject);
object.channel = channel;
object.type = objectType;
object.id = objectId;
object.spec = specId;
````

However[ in `setItem`](https://github.com/oddnetworks/oddworks/blob/master/lib/services/catalog/commands/index.js#L13-L24) we are looking for `object.source`.   

Problem:  it isn't there.

## Possible Solution

I'm, not sure here...
(1) The source object isn't coming out of the store with `source` in it's keys.  Maybe we staple that to the object before we send it to `setItem`?

or...
(2) Do we change `setItem` to look for `.spec` rather than `.source`?

**Comment by aljachimiak** [comment]
With a little github digging, this may be a copy/paste error.

Previous versions never looked for `.source`.

For example: [https://github.com/oddnetworks/oddworks/blob/3.10.0/lib/services/catalog/commands/index.js#L16-L21](https://github.com/oddnetworks/oddworks/blob/3.10.0/lib/services/catalog/commands/index.js#L16-L21)

**Comment by aljachimiak** [comment]
@kixxauth fixed this in #217, with [this commit](https://github.com/oddnetworks/oddworks/commit/88fdb61a688616e2bd495e438e02820555a0688f).

Closed.
